<?xml version="10.0"?>
<launch>
<!-- LiDAR topic -->
	<param name="lidar_topic" value="/velodyne_points" />
<!-- IMU topic -->
	<param name="IMU_topic" value="/NA_nav_ned" />
	
	<!--
	# AUTHOR: CALEB WILSON (CALEBWILSON@fau.edu or cwilson15@lssu.edu)
	# This is the launch file for the vision system for the WAMV-16 at FAU
	# Date: 06-13-2023
	 -->
<!-- 4.0 SENSOR DRIVER -->
   <!-- 4.1 ZED 2i LAUNCH -->
	<include file="$(find zed_wrapper)/launch/zed2i.launch" />
	
   <!-- 4.2 VELODYNE LAUNCH -->
	<include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch" />
<!--
###################################ROS PARAMETERS#######################################
-->
 	<param name="lidar_offset_x" value="-.5" /> <!-- lidar offset in the x (forward dimension) from center of boat in m-->
	<param name="lidar_offset_y" value="0" /> <!-- lidar offset in the x (sideways dimension) from center of boat in m-->
	<param name="lidar_offset_z" value="-.1" /> <!-- lidar offset in the z (vertical dimension) from center of boat in m-->
	<param name="lidar_range" value="40" /> <!-- range in m  -->
	<param name="occ_grid_range" value="50" /> <!-- range in m  -->
	<param name="grid_size" value=".5" /> <!-- grid_size in m  -->
	<param name="lidar_sensitivity" value="120" /> <!-- [1-100]% sensitivity  -->
	<param name="lidar_intensity_l" value="0.0" /> <!-- set between for the intensity low 0-100 -->
	<param name="lidar_intensity_h" value="100.0" /> <!-- set between 0-100 for the intensity high -->
	<param name="lidar_model" value="16" /> <!-- set between 0-100 -->
<!-- 2.0 VEHICLE PARAMETERS -->
	<param name="vehicle_dim_length" value="1" /> <!-- this is the length of the boat in m-->
	<param name="vehicle_dim_width" value="2" /> <!-- this is the width of the boat  in m-->
	<param name="vehicle_lidar_height" value="1.5" /> <!-- this is the height of the lidar from the aters surface-->
<!-- 1.1 Ransac Parameters for initial passthrough filter-->	
 	<param name="PASS_L" value="0" /> <!--passthrough filter ... default 0  -->
 	<param name="PASS_H" value="1.5" /> <!-- passthrough filter ... default 1.5  -->
<!-- 1.2 Ransac Parameters for finding Normals--> 	
	<param name="NORM_K_SEARCH" value="50" /> <!-- estimates point noramls ... default 50  -->
<!-- 1.3 Ransac Parameters for finding planar segmentation--> 	
	<param name="SEG_DIST_WEIGHT" value=".01" /> <!-- weight for noramls...default .01 -->
	<param name="SEG_MAX_ITERATIONS" value="100" /> <!-- default ... 100  -->
	<param name="SEG_DIST_THRESH" value=".01" /> <!--  ... default .01  -->
<!-- 1.4 Ransac Parameters for finding Cylinder--> 	
	<param name="CYL_DIST_WEIGHT" value=".1" /> <!-- weight for noramls...default .1 -->
	<param name="CYL_MAX_ITERATIONS" value="500" /> <!-- default ... 500  -->
	<param name="CYL_DIST_THRESH" value=".01" /> <!--  ... default .01  -->
	<param name="CYL_RADIUS_L" value=".2" /> <!--  ... default .2  -->
	<param name="CYL_RADIUS_H" value=".4" /> <!--  ... default .4  --> 
  	<!--
	#############################SCRIPT DECLARATION###################  	
  	 -->	
 	<node pkg="fau_vision" type="tf_broadcast" name="tf_broadcast" output="screen">
  	</node>
  	<!-- -->
	<node pkg="fau_vision" type="PCL_TEST" name="PCL_TEST" output="screen">
  	</node>
  	<!-- -->  	
	<node pkg="fau_vision" type="Voxel_Grid" name="Voxel_Grid" output="screen">
  	</node>
  	<!-- -->  	
	<node pkg="fau_vision" type="Statistical_Outlier" name="Statistical_Outlier" output="screen">
  	</node>
  	<!-- --> 	
	<node pkg="fau_vision" type="Ransac" name="Ransac" output="screen">
  	</node>
  	<!-- 
	<node pkg="fau_vision" type="cylinder" name="cylinder" output="screen">
  	</node> -->
  	<!-- -->  	
	<node pkg="fau_vision" type="transform_3d" name="transform_3d" output="screen">
  	</node>
  	<!--   	
	<node pkg="fau_vision" type="class_test.py" name="class_test" output="screen">
  	</node> -->  	
  	
<!-- X.0 LAUNCHES THE LAUNCH FILE -->	
</launch>
